src/components/Cluster/ClusterDetail/__tests__/UpgradeClusterModal.tsx

```diff
@@ -0,0 +1,261 @@
+import { fireEvent, screen } from '@testing-library/react';
+import UpgradeClusterModal from 'Cluster/ClusterDetail/UpgradeClusterModal';
+import React from 'react';
+import { IState } from 'reducers/types';
+import { v5ClusterResponse } from 'testUtils/mockHttpCalls';
+import { renderWithStore } from 'testUtils/renderUtils';
+
+function renderAndOpen(
+  props: React.ComponentPropsWithoutRef<typeof UpgradeClusterModal> = {},
+  state: IState = {}
+) {
+  interface IComponent {
+    show: () => void;
+  }
+
+  const elementRef = React.createRef<IComponent>();
+  const defaultProps = Object.assign({}, { ref: elementRef }, props);
+  const element = renderWithStore(UpgradeClusterModal, defaultProps, state);
+
+  // Make modal visible.
+  (elementRef.current as IComponent).show();
+
+  return element;
+}
+
+function createRelease(version: string, active: boolean): IRelease {
+  return {
+    version,
+    active,
+    timestamp: '2020-06-11T12:34:56Z',
+    components: [{ name: 'kubernetes', version: '1.16.3' }],
+    changelog: [{ component: 'dummy', description: 'dummy' }],
+    kubernetesVersion: '1.16.3',
+    releaseNotesURL: 'dummy',
+  } as IRelease;
+}
+
+function createInitialState(
+  releases: Record<string, IRelease>,
+  isAdmin: boolean = false
+) {
+  return {
+    entities: {
+      releases: {
+        items: releases,
+      },
+    },
+    main: {
+      loggedInUser: {
+        isAdmin,
+      },
+    },
+  };
+}
+
+describe('UpgradeClusterModal', () => {
+  it('renders without crashing', () => {
+    const cluster = Object.assign({}, v5ClusterResponse, {
+      capabilities: {},
+    });
+    renderAndOpen({
+      cluster: cluster,
+    });
+  });
+
+  it('renders the inspect changes page by default', () => {
+    const cluster = Object.assign({}, v5ClusterResponse, {
+      capabilities: {},
+    });
+    renderAndOpen({
+      cluster: cluster,
+    });
+
+    expect(
+      screen.getByText(/Before upgrading please acknowledge the following/i)
+    ).toBeInTheDocument();
+  });
+
+  it(`can't change the upgrade release version if the user is not an admin`, () => {
```

Comment:
More specifically, Is this test something like:

`can't change the upgrade release version if the user is not an admin, and there are no active releases in the state`

Reply:
The fact that there are no active releases in the store doesn't make a difference. This just checks that the `Change version` button is not there if the user is not an admin